src/core/Tools.cpp

```diff
@@ -296,16 +299,80 @@ namespace Tools
         return true;
     }
 
-    // Escape regex symbols
-    auto regexEscape = QRegularExpression(R"re(([-[\]{}()+.,\\\/^$#|*?]))re");
+    /****************************************************************************
+     *
+     * Copyright (C) 2020 Giuseppe D'Angelo <dangelog@gmail.com>.
+     * Copyright (C) 2020 Klarälvdalens Datakonsult AB, a KDAB Group company, info@kdab.com, author Giuseppe D'Angelo
+     * <giuseppe.dangelo@kdab.com> Copyright (C) 2021 The Qt Company Ltd. Contact: https://www.qt.io/licensing/
+     *
+     * This function is part of the QtCore module of the Qt Toolkit. And subject to the following licenses.
+     *
+     * Commercial License Usage
+     * Licensees holding valid commercial Qt licenses may use this file in
+     * accordance with the commercial license agreement provided with the
+     * Software or, alternatively, in accordance with the terms contained in
+     * a written agreement between you and The Qt Company. For licensing terms
+     * and conditions see https://www.qt.io/terms-conditions. For further
+     * information use the contact form at https://www.qt.io/contact-us.
+     *
+     * GNU Lesser General Public License Usage
+     * Alternatively, this file may be used under the terms of the GNU Lesser
+     * General Public License version 3 as published by the Free Software
+     * Foundation and appearing in the file LICENSE.LGPL3 included in the
+     * packaging of this file. Please review the following information to
+     * ensure the GNU Lesser General Public License version 3 requirements
+     * will be met: https://www.gnu.org/licenses/lgpl-3.0.html.
+     *
+     * GNU General Public License Usage
+     * Alternatively, this file may be used under the terms of the GNU
```

Comment:
```suggestion
     * Alternatively, this file may be used under the terms of the GNU
```
```suggestion
     * This function may be used under the terms of the GNU
```

Overall this can be confusing. What about leaving the license text, and just mentioning here copyright lines and the fact that this was adapted from Qt source.

Then in the `COPYING` file, add a new block linking to this?